sway-lsp/src/utils/function.rs

```diff
@@ -1,6 +1,6 @@
-use sway_core::FunctionDeclaration;
+use sway_types::Span;
 
-pub(crate) fn extract_fn_signature(func_dec: &FunctionDeclaration) -> String {
-    let value = func_dec.span.as_str();
+pub(crate) fn extract_fn_signature(span: &Span) -> String {
```

Comment:
If we change this to take a span directly instead of a `FunctionDeclaration`, we should add a doc comment to the function that states what assumptions we make about the given `Span`. E.g. it expects a span from either a `FunctionDeclaration`, trait function, or whatever other declarations this is used for. Should probably do the same for `get_function_details` too.

Reply:
Added comments